sources/packages/backend/libs/integrations/src/esdc-integration/e-cert-integration/e-cert-file-handler.ts

```diff
@@ -99,97 +108,123 @@ export class ECertFileHandler extends ESDCFileHandler {
   }
 
   /**
-   * Get all Full-Time/Part-Time disbursements available to be sent to ESDC.
+   * Generate the e-Cert file for Full-Time/Part-Time disbursements available to be sent to ESDC.
    * Consider any record that is scheduled in upcoming days or in the past.
-   * @param eCertIntegrationService
+   * @param eCertIntegrationService Full-Time/Part-Time integration responsible
+   * for the respective integration.
    * @param offeringIntensity disbursement offering intensity.
-   * @param fileCode File code applicable for Part-Time or Full-Time.
-   * @param sequenceGroup Sequence group application for Part-Time or Full-Time.
+   * @param fileCode file code applicable for Part-Time or Full-Time.
+   * @param sequenceGroupPrefix sequence group prefix for Part-Time or Full-Time
+   * file sequence generation.
    * @returns result of the file upload with the file generated and the
    * amount of records added to the file.
    */
   private async generateECert(
     eCertIntegrationService: ECertIntegrationService,
     offeringIntensity: OfferingIntensity,
     fileCode: string,
-    sequenceGroup: string,
+    sequenceGroupPrefix: string,
   ): Promise<ECertUploadResult> {
     this.logger.log(
       `Retrieving ${offeringIntensity} disbursements to generate the e-Cert file...`,
     );
-    const disbursements =
-      await this.disbursementScheduleService.getECertInformationToBeSent(
-        offeringIntensity,
-      );
-    if (!disbursements) {
-      return {
-        generatedFile: "none",
-        uploadedRecords: 0,
-      };
-    }
-    this.logger.log(
-      `Found ${disbursements.length} ${offeringIntensity} disbursements schedules.`,
-    );
-    const disbursementRecords = disbursements.map((disbursement) => {
-      return this.createECertRecord(disbursement);
-    });
-
-    // Fetches the disbursements ids, for further update in the DB.
-    const disbursementIds = disbursements.map(
-      (disbursement) => disbursement.id,
-    );
-
-    //Create records and create the unique file sequence number.
-    let uploadResult: ECertUploadResult;
-    const now = new Date();
-    await this.sequenceService.consumeNextSequence(
-      `${sequenceGroup}_${getISODateOnlyString(new Date())}`,
-      async (nextSequenceNumber: number, entityManager: EntityManager) => {
-        try {
-          this.logger.log(
-            `Creating  ${offeringIntensity} e-Cert file content...`,
-          );
-          const fileContent = eCertIntegrationService.createRequestContent(
-            disbursementRecords,
+    try {
+      const sequenceGroup = `${sequenceGroupPrefix}_${getISODateOnlyString(
```

Comment:
why the date is appended to the prefix

Reply:
oh... is it to reset the generate sequence each day?

Reply:
Yes, it is. Not new in this PR, the logic is the same for a while 😉